### PR TITLE
feat(ui): add MenuV2

### PR DESCRIPTION
**Pull request by QuiiBz**
## Type

- Feature

#### What is expected?

Add a new `MenuV2` component that uses the internal `Popup` component to replace the usage of `reakit`. We have to introduce a new component and deprecate the old one because the types are a bit different, especially the `placement` prop.

#### The following changes where made:

1. Add `MenuV2` component with tests and stories similar to `Menu`

2. Deprecated `Menu` component

3. Add migration doc

### CODE REVIEW

**Comment by QuiiBz** [comment]
Needs https://github.com/scaleway/ultraviolet/pull/2920

File: packages/ui/src/components/MenuV2/index.tsx [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+      // Focus the disclosure when the menu is closed
+      disclosureRef.current?.focus()
+    }
+
+    const finalDisclosure = cloneElement(target, {
+      onClick: toggleVisible,
+      'aria-haspopup': 'dialog',
+      'aria-expanded': isVisible,
+      // @ts-expect-error not sure how to fix this
```
Comment (philibea) [inline packages/ui/src/components/MenuV2/index.tsx:137]: why ?

File: packages/ui/src/components/MenuV2/index.tsx [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+      placement = 'bottom',
+      visible = false,
+      className,
+      'data-testid': dataTestId,
+    }: MenuProps,
+    ref: Ref<HTMLButtonElement | null>,
+  ) => {
+    const [isVisible, setIsVisible] = useState(visible)
+    const popupRef = useRef<HTMLDivElement>(null)
+    const disclosureRef = useRef<HTMLDivElement>(null)
```
Comment (philibea) [inline packages/ui/src/components/MenuV2/index.tsx:99]: disclosure should not be an htmlDivElement ? It's always a Button ?

File: packages/ui/src/components/MenuV2/index.tsx [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    const onClose = () => {
+      setIsVisible(false)
+
+      // Focus the disclosure when the menu is closed
+      disclosureRef.current?.focus()
+    }
+
+    const finalDisclosure = cloneElement(target, {
+      onClick: toggleVisible,
+      'aria-haspopup': 'dialog',
```
Comment (philibea) [inline packages/ui/src/components/MenuV2/index.tsx:135]:
```suggestion
      'aria-haspopup': 'menu',
```

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-haspopup

Comment (QuiiBz) [reply packages/ui/src/components/MenuV2/index.tsx:135]: That's not what reakit uses:

Screenshot 2023-09-18 at 09 38 27

File: packages/ui/src/components/MenuV2/index.tsx (hunk 1 above)
Comment (QuiiBz) [reply packages/ui/src/components/MenuV2/index.tsx:137]: `cloneElement` doesn't seem to accept `ref` as a prop

**Review by matthprost — CHANGES_REQUESTED** [review]

**Review by matthprost — APPROVED** [review]

File: packages/ui/src/components/MenuV2/Item.tsx [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+      className,
+      'data-testid': dataTestId,
+    },
+    ref,
+  ) => {
+    if (href && !disabled) {
+      return (
+        <Tooltip text={tooltip}>
+          <StyledLinkItem
+            borderless
```
Comment (ModuloM) [inline packages/ui/src/components/MenuV2/Item.tsx:112]: Do we always want that Link Items are borderless?

File: packages/ui/src/components/MenuV2/__stories__/Sentiments.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import type { StoryFn } from '@storybook/react'
+import { MenuV2 } from '..'
+import { DefaultDisclosure } from './Template.stories'
+
+export const Sentiments: StoryFn<typeof MenuV2> = () => (
+  <MenuV2 disclosure={DefaultDisclosure}>
+    <MenuV2.Item>default disabled</MenuV2.Item>
+    <MenuV2.Item sentiment="danger">Danger disabled</MenuV2.Item>
```
Comment (ModuloM) [inline packages/ui/src/components/MenuV2/__stories__/Sentiments.stories.tsx:8]: You've switch disable texts for 'default' and 'danger' is this examples

File: packages/ui/src/components/MenuV2/__stories__/WithModal.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+      <div style={{ padding: 32 }}>
+        Content should be present in center of the modal
+      </div>
+    </Modal>
+  )
+
+  return (
+    <MenuV2 disclosure={DefaultDisclosure}>
+      <MenuV2.Item>Menu Item</MenuV2.Item>
+      <MenuV2.Item href="/?path=/docs/components-navigation-menu--modal">
```
Comment (ModuloM) [inline packages/ui/src/components/MenuV2/__stories__/WithModal.stories.tsx:21]: It's not very important but the link seems broken

`/?/?path=/docs/components-navigation-menu"`

File: packages/ui/src/components/MenuV2/index.tsx [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+    }
+
+    const onClose = () => {
+      setIsVisible(false)
+
+      // Focus the disclosure when the menu is closed
+      disclosureRef.current?.focus()
+    }
+
+    const finalDisclosure = cloneElement(target, {
```
Comment (ModuloM) [inline packages/ui/src/components/MenuV2/index.tsx:133]: 👍

**Review by ModuloM — APPROVED** [review]

File: packages/ui/src/components/MenuV2/__stories__/Sentiments.stories.tsx (hunk 5 above)
Comment (QuiiBz) [reply packages/ui/src/components/MenuV2/__stories__/Sentiments.stories.tsx:8]: It's only borderless when there is an `href`, similar to `Menu` so I guess it's fine.

File: packages/ui/src/components/MenuV2/Item.tsx (hunk 4 above)
Comment (QuiiBz) [reply packages/ui/src/components/MenuV2/Item.tsx:112]: It's only borderless when there is an href, similar to Menu so I guess it's fine.

**Pull request merged**